app/preprints/-components/submit/supplements/component.ts

```diff
@@ -11,24 +13,69 @@ interface SupplementsArgs {
     manager: PreprintStateMachine;
 }
 
+interface SupplementsForm{
+    node: Node;
+}
+
+const SupplementFormValidation: ValidationObject<SupplementsForm> = {
+    node: validatePresence({
+        presence: true,
+        ignoreBlank: true,
+        type: 'empty',
+    }),
+};
+
 /**
  * The Supplements Component
  */
 export default class Supplements extends Component<SupplementsArgs>{
-    @service intl!: Intl;
+    @tracked displayExistingNodeWidget = false;
+    @tracked displayCreateNodeWidget = false;
+    @tracked displayCancelButton = false;
+
+    supplementFormChangeset = buildChangeset(this.args.manager.preprint, SupplementFormValidation);
+
+    private isDisplayCancelButton(): void {
+        this.displayCancelButton = this.displayExistingNodeWidget ||
+        this.displayCreateNodeWidget;
+    }
+
+    @action
+    public onCancelProjectAction(): void {
+        this.displayExistingNodeWidget = false;
+        this.displayCreateNodeWidget = false;
+        this.isDisplayCancelButton();
+    }
 
     @action
     public onConnectOsfProject(): void {
-        this.validate();
+        this.displayExistingNodeWidget = true;
+        this.displayCreateNodeWidget = false;
+        this.isDisplayCancelButton();
     }
 
     @action
     public onCreateOsfProject(): void {
+        this.displayCreateNodeWidget = true;
+        this.displayExistingNodeWidget = false;
+        this.isDisplayCancelButton();
+    }
+
+    @action
+    public selectNode(node: Node): void {
+        this.supplementFormChangeset.set('node', node);
         this.validate();
     }
```

Comment:
Again, optionally, we could just set the `node` relationship on the preprint directly without having to go through the changeset, if we forego a changeset on this page.